pylib/gyp/generator/ninja.py

```diff
@@ -707,14 +707,14 @@ def WriteRules(
                 rule.get("message", None),
                 ("%s " + generator_default_variables["RULE_INPUT_PATH"]) % name,
             )
-            is_cygwin = (
-                self.msvs_settings.IsRuleRunUnderCygwin(rule)
-                if self.flavor == "win"
-                else False
+            win_shell_flags = (
+                self.msvs_settings.GetRuleShellFlags(action)
```

Comment:
Is `action` should be `rule` ?

Reply:
It is `action` that should be passed here, I'll rename `GetRuleShellFlags` so it is not confusing.

Reply:
After looking at the code more closely I realized it is indeed `rule` that should be passed.